src/tools/rbd_mirror/ImageReplayer.cc

```diff
@@ -970,7 +971,13 @@ void ImageReplayer<I>::handle_shut_down(int r) {
     auto ctx = new LambdaContext([this, r](int) {
         handle_shut_down(r);
       });
-    m_local_status_updater->remove_mirror_image_status(m_global_image_id, ctx);
+    if (m_finished || m_delete_in_progress) {
+      m_local_status_updater->remove_mirror_image_status(m_global_image_id,
+                                                         true, ctx);
```

Comment:
Just wondering, do we really need to remove the local status here? I would expect that all local structures would be removed when the mirroring is disabled (e.g. as one of the remove image steps). If it did not reliably work there for some reason then I think we would better fix it.

Reply:
Yes indeed just tested and this is not needed!

Reply:
I had to re-add this. This is needed in certain cases like with rbd-mirror daemons on a single cluster.